prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,205 @@
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for creating a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representing a Kubernetes V1Job specification.
+        namespace: The Kubernetes namespace to create this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return api_client.create_namespaced_job(
+        namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def delete_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+    delete_option_kwargs: Optional[Dict] = None,
+) -> V1Status:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to delete.
+        namespace: The Kubernetes namespace to delete this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+        - delete_option_kwargs: Optional keyword arguments to pass to
+            the V1DeleteOptions object (e.g. {"propagation_policy": "...",
+            "grace_period_seconds": "..."}.
+
+    Returns:
+        A Kubernetes `V1Status` object.
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    if delete_option_kwargs:
+        kube_kwargs.update(body=client.V1DeleteOptions(**delete_option_kwargs))
+
+    return api_client.delete_namespaced_job(
+        name=job_name, namespace=namespace, **kube_kwargs
+    )
+
+
+@task
+async def list_namespaced_job(
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1JobList:
+    """Task for listing namespaced Kubernetes jobs.
+
+    Args:
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        namespace: The Kubernetes namespace to list jobs from.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Returns:
+        A Kubernetes `V1JobList` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.list_namespaced_job(
```

Comment:
I think we need to call `await run_sync_on_thread(api_client.list_namespaced_job, namespace=namespace, **kube_kwargs)` for all these calls if I am not mistaken.

Reply:
why?

Reply:
To generalize from my understanding, if the client makes any network calls, it should be done asynchronously in our Collections, whether through the built-in async capabilities or our "hack" of running it as a separate thread.